internal/services/workloads/workloads_sap_virtual_instance.go

```diff
@@ -0,0 +1,1001 @@
+		DatabaseServer:    flattenDatabaseServerFullResourceNames(input.DatabaseServer),
+		SharedStorage:     flattenSharedStorage(input.SharedStorage),
+	}
+
+	return append(result, fullResourceNames)
+}
+
+func flattenApplicationServerFullResourceNames(input *sapvirtualinstances.ApplicationServerFullResourceNames) []ApplicationServerFullResourceNames {
+	if input == nil {
+		return nil
+	}
+
+	result := make([]ApplicationServerFullResourceNames, 0)
+
+	applicationServerFullResourceNames := ApplicationServerFullResourceNames{
+		VirtualMachines: flattenVirtualMachinesFullResourceNames(input.VirtualMachines),
+	}
+
+	if v := input.AvailabilitySetName; v != nil {
+		applicationServerFullResourceNames.AvailabilitySetName = *v
+	}
+
+	return append(result, applicationServerFullResourceNames)
+}
+
+func flattenVirtualMachinesFullResourceNames(input *[]sapvirtualinstances.VirtualMachineResourceNames) []VirtualMachineFullResourceNames {
+	if input == nil {
+		return nil
+	}
+
+	result := make([]VirtualMachineFullResourceNames, 0)
+
+	for _, item := range *input {
+		virtualMachineFullResourceNames := VirtualMachineFullResourceNames{
+			DataDiskNames:         flattenDataDiskNames(item.DataDiskNames),
+			NetworkInterfaceNames: flattenNetworkInterfaceResourceNames(item.NetworkInterfaces),
+		}
+
+		if v := item.HostName; v != nil {
+			virtualMachineFullResourceNames.HostName = *v
+		}
+
+		if v := item.OsDiskName; v != nil {
+			virtualMachineFullResourceNames.OSDiskName = *v
+		}
+
+		if v := item.VirtualMachineName; v != nil {
+			virtualMachineFullResourceNames.VMName = *v
+		}
+
+		result = append(result, virtualMachineFullResourceNames)
+	}
+
+	return result
+}
+
+func flattenCentralServerFullResourceNames(input *sapvirtualinstances.CentralServerFullResourceNames) []CentralServerFullResourceNames {
+	if input == nil {
+		return nil
+	}
+
+	result := make([]CentralServerFullResourceNames, 0)
+
+	centralServerFullResourceNames := CentralServerFullResourceNames{
+		LoadBalancer:    flattenLoadBalancerFullResourceNames(input.LoadBalancer),
+		VirtualMachines: flattenVirtualMachinesFullResourceNames(input.VirtualMachines),
+	}
+
+	if v := input.AvailabilitySetName; v != nil {
+		centralServerFullResourceNames.AvailabilitySetName = *v
+	}
+
+	return append(result, centralServerFullResourceNames)
+}
+
+func flattenLoadBalancerFullResourceNames(input *sapvirtualinstances.LoadBalancerResourceNames) []LoadBalancer {
+	if input == nil {
+		return nil
+	}
+
+	result := make([]LoadBalancer, 0)
+	loadBalancer := LoadBalancer{}
+
+	if v := input.LoadBalancerName; v != nil {
+		loadBalancer.Name = *v
+	}
+
+	if v := input.BackendPoolNames; v != nil {
+		loadBalancer.BackendPoolNames = *v
+	}
+
+	if v := input.FrontendIPConfigurationNames; v != nil {
+		loadBalancer.FrontendIpConfigurationNames = *v
+	}
+
+	if v := input.HealthProbeNames; v != nil {
+		loadBalancer.HealthProbeNames = *v
+	}
```

Comment:
We can replace many of these with the `pointer.From` function 
```suggestion
	loadBalancer.Name = pointer.From(input.LoadBalancerName)
	
	...etc
```
Please go through the all the flatten functions and replace these where possible.

Reply:
Updated